macros/edr/data_monitoring/data_monitors_configuration/get_buckets_configuration.sql

```diff
@@ -64,40 +64,40 @@
-                when last_min_bucket_end > days_back_start then days_back_start {# When the metric was collected recently, but for a period that is smaller than days_back #}
-                when last_max_bucket_end < backfill_start then last_max_bucket_end {# When the metric was not collected for a period longer than backfill_days #}
-                else {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, 'periods_to_backfill', 'last_max_bucket_end')) }} {# When backfill reduce full time buckets from last_max_bucket_end to backfill #}
+                when max_existing_bucket_end is null then days_back_start {# When this is the first run of this metric #}
+                when max_existing_bucket_end <= days_back_start then days_back_start {# When the metric was not collected for a period longer than days_back #}
+                when min_existing_bucket_start > days_back_start then days_back_start {# When the metric was collected recently, but for a period that is smaller than days_back #}
+                when max_existing_bucket_end <= backfill_start then max_existing_bucket_end {# When the metric was not collected for a period longer than backfill_days #}
+                else {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, 'periods_to_backfill', 'max_existing_bucket_end')) }} {# When backfill reduce full time buckets from max_existing_bucket_end to backfill #}
             end as min_bucket_start,
             case
                 {# This makes sure we collect only full bucket #}
-                when last_max_bucket_end is null or last_max_bucket_end < days_back_start or last_min_bucket_end > days_back_start
+                when max_existing_bucket_end is null or max_existing_bucket_end <= days_back_start or min_existing_bucket_start > days_back_start
                 then {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, 'periods_until_max', 'days_back_start')) }} {# Add full buckets to days_back_start #}
-                else {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, 'periods_until_max', 'last_max_bucket_end')) }} {# Add full buckets to last_max_bucket_end #}
+                else {{ elementary.edr_cast_as_timestamp(elementary.edr_timeadd(metric_properties.time_bucket.period, 'periods_until_max', 'max_existing_bucket_end')) }} {# Add full buckets to max_existing_bucket_end #}
             end as max_bucket_end
         from full_buckets_calc
     {%- endset %}
 
     {# We assume we should also cosider sources as incremental #}
-    {% if not (elementary.is_incremental_model(elementary.get_model_graph_node(model), source_included=true) or unit_test) %}
+    {% if not (elementary.is_incremental_model(elementary.get_model_graph_node(model_relation), source_included=true) or unit_test) %}
```

Comment:
This is something that broke a while ago (I think due to my changes), that actually caused us to always return False from `is_incremental_model`